osu.Game/Screens/SelectV2/UpdateBeatmapSetButtonV2.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Diagnostics;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Overlays;
+using osu.Game.Screens.Select.Carousel;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.SelectV2
+{
+    public partial class UpdateBeatmapSetButtonV2 : OsuAnimatedButton
+    {
+        private BeatmapSetInfo? beatmapSet;
+
+        public BeatmapSetInfo? BeatmapSet
+        {
+            get => beatmapSet;
+            set
+            {
+                beatmapSet = value;
+
+                if (IsLoaded)
+                    beatmapChanged();
```

Comment:
I'm not sure this is doing what you expect it to? It doesn't seem to be updating anything when the set changes. It will still be tracking the previous set. 😅

Reply:
I must be missing something here, this function is a callback whenever `beatmapSet` changes, and that function updates the display state of the button. There's nothing else to update..?

Reply:
You might be right on this one. I thought a hook-up was required to the downloader, but seems not.